diff --git a/Products/zms/zmslinkelement.py b/Products/zms/zmslinkelement.py
index b031133e..8d1bb936 100644
--- a/Products/zms/zmslinkelement.py
+++ b/Products/zms/zmslinkelement.py
@@ -102,8 +102,9 @@ class ZMSLinkElement(zmscustom.ZMSCustom):
     # --------------------------------------------------------------------------
     #  ZMSLinkElement.getEmbedType: 
     # --------------------------------------------------------------------------
-    def getEmbedType(self):
-      embed_type = self.getObjAttrValue( self.getObjAttr( 'attr_type'), self.REQUEST)
+    def getEmbedType(self, **kwargs):
+      request = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
+      embed_type = self.getObjAttrValue( self.getObjAttr( 'attr_type'), request)
       if embed_type in [ 'embed', 'recursive', 'remote']:
         ref_obj = self.getRefObj()
         if ref_obj is not None and ref_obj.isAncestor( self):
@@ -365,7 +366,7 @@ class ZMSLinkElement(zmscustom.ZMSCustom):
     # --------------------------------------------------------------------------
     #  ZMSLinkElement.isPage
     # --------------------------------------------------------------------------
-    def isPage(self):
+    def isPage(self, **kwargs):
       rtnVal = False
       if self.getEmbedType() == 'remote':
         return self.getRemoteObj().get('is_page',False)
@@ -391,7 +392,7 @@ class ZMSLinkElement(zmscustom.ZMSCustom):
             rtnVal = rtnVal or ref_obj.isPageElement()
           else:
             rtnVal = rtnVal or True
-        elif self.getObjProperty('align', self.REQUEST) not in ['', 'NONE']:
+        elif self.getObjProperty('align', self.get('REQUEST', {})) not in ['', 'NONE']:
           rtnVal = rtnVal or True
       return rtnVal
 
@@ -628,9 +629,9 @@ class ZMSLinkElement(zmscustom.ZMSCustom):
     #
     #  Returns self or referenced object (if embedded) as ZMSProxyObject
     # --------------------------------------------------------------------------
-    def __proxy__(self):
+    def __proxy__(self, **kwargs):
       rtn = self
-      req = self.REQUEST
+      req = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
       if req.get( 'ZMS_PROXY', True):
         if req.get( 'URL', '').find( '/manage') < 0 or req.get( 'ZMS_PATH_HANDLER', False):
           if self.isEmbeddedRecursive():
@@ -647,9 +648,9 @@ class ZMSLinkElement(zmscustom.ZMSCustom):
     #  Returns self or proxy-object from Path-Handler (if embedded) as 
     #  ZMSProxyObject.
     # --------------------------------------------------------------------------
-    def getProxy(self):
+    def getProxy(self, **kwargs):
       rtn = self
-      req = self.REQUEST
+      req = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
       if req.get( 'ZMS_PROXY', True):
         rtn = req.get( 'ZMS_PROXY_%s'%self.id, self.__proxy__())
       return rtn
