diff --git a/Products/zms/_cachemanager.py b/Products/zms/_cachemanager.py
index 1de23ede..88aadcd8 100644
--- a/Products/zms/_cachemanager.py
+++ b/Products/zms/_cachemanager.py
@@ -39,8 +39,8 @@ class ReqBuff(object):
     #
     #  Clear buffered values from Http-Request.
     # --------------------------------------------------------------------------
-    def clearReqBuff(self, prefix='', REQUEST=None):
-      request = self.REQUEST
+    def clearReqBuff(self, prefix='', **kwargs):
+      request = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
       buff = request.get('__buff__', Buff())
       reqBuffId = self.getReqBuffId(prefix)
       if len(prefix) > 0:
@@ -56,8 +56,8 @@ class ReqBuff(object):
     #
     #  @throws Exception
     # --------------------------------------------------------------------------
-    def fetchReqBuff(self, key, REQUEST=None):
-      request = self.REQUEST
+    def fetchReqBuff(self, key, **kwargs):
+      request = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
       buff = request['__buff__']
       reqBuffId = self.getReqBuffId(key)
       return getattr(buff, reqBuffId)
@@ -67,14 +67,14 @@ class ReqBuff(object):
     #
     #  Returns value and stores it in buffer of Http-Request.
     # --------------------------------------------------------------------------
-    def storeReqBuff(self, key, value, REQUEST=None):
-      request = self.REQUEST
+    def storeReqBuff(self, key, value, **kwargs):
+      request = kwargs.get('request', kwargs.get('REQUEST', self.get('REQUEST', {})))
       buff = request.get('__buff__', None)
       if buff is None:
         buff = Buff()
       reqBuffId = self.getReqBuffId(key)
       setattr(buff, reqBuffId, value)
-      request.set('__buff__', buff)
+      request['__buff__'] = buff
       return value
 
 ################################################################################
