diff --git a/Products/zms/_xmllib.py b/Products/zms/_xmllib.py
index 055b3be9..1cbd3327 100644
--- a/Products/zms/_xmllib.py
+++ b/Products/zms/_xmllib.py
@@ -645,11 +645,11 @@ def getObjPropertyToXml(self, REQUEST, base_path='', data2hex=False, obj_attr={}
     lang = REQUEST.get('lang')
     langIds = self.getLangIds()
     for langId in langIds:
-      REQUEST.set('lang', langId)
+      REQUEST['lang'] = langId
       s_attr_xml = getAttrToXml(self, base_path, data2hex, obj_attr, REQUEST)
       if len(s_attr_xml) > 0:
         xml += '<lang%s>%s</lang>' % ( [ '' , ' id="%s"'%langId ][ int(len(langIds)>1) ] , s_attr_xml )
-    REQUEST.set('lang', lang)
+    REQUEST['lang'] = lang
   # Simple Attributes.
   else:
     xml += getAttrToXml(self, base_path, data2hex, obj_attr, REQUEST)
