diff --git a/Products/zms/zmslog.py b/Products/zms/zmslog.py
index 85469ed0..0c869287 100644
--- a/Products/zms/zmslog.py
+++ b/Products/zms/zmslog.py
@@ -115,7 +115,9 @@ class ZMSLog(ZMSItem.ZMSItem):
     # --------------------------------------------------------------------------
     def getLOG(self, REQUEST, RESPONSE=None):
       """ ZMSLog.getLOG """
-      filename = os.path.join(standard.getINSTANCE_HOME(),'var','log','event.log')
+      filename = os.path.join(standard.getINSTANCE_HOME(), 'log', 'zms-event.log')
+      if not os.path.exists(filename):
+        filename = os.path.join(standard.getINSTANCE_HOME(), 'var', 'log', 'event.log')
       RESPONSE.setHeader( 'Content-Type','text/plain')
       RESPONSE.setHeader( 'Content-Disposition','inline;filename="%s"'%_fileutil.extractFilename( filename))
       file = open( filename, 'r')
@@ -127,7 +129,9 @@ class ZMSLog(ZMSItem.ZMSItem):
     #  ZMSLog.tail_event_log:
     # --------------------------------------------------------------------------
     def tail_event_log(self, linesback=100, returnlist=True):
-      filename = os.path.join(standard.getINSTANCE_HOME(),'var','log','event.log')
+      filename = os.path.join(standard.getINSTANCE_HOME(), 'log', 'zms-event.log')
+      if not os.path.exists(filename):
+        filename = os.path.join(standard.getINSTANCE_HOME(), 'var', 'log', 'event.log')
       try:
         return _fileutil.tail_lines(filename,linesback,returnlist)
       except:
