diff --git a/Products/zms/zmscustom.py b/Products/zms/zmscustom.py
index 8f1bdf61..926b150f 100644
--- a/Products/zms/zmscustom.py
+++ b/Products/zms/zmscustom.py
@@ -412,9 +412,31 @@ class ZMSCustom(zmscontainerobject.ZMSContainerObject):
             values = {}
             for dictitem in listitem['dictionary']['item']:
               key = dictitem.get('@key')
+              typ = dictitem.get('@type')
+              val = dictitem.get('#text')
+              # filter out internal attributes not to be exposed
+              if key in ('_change_dt', '_change_uid', '_grid'):
+                  continue
               if key not in keys:
                 keys.append(key)
-              values[key] = dictitem.get('#text')
+              if typ == 'list' and dictitem.get('list') is not None:
+                items = dictitem.get('list').get('item')
+                if type(items) is list:
+                  val = ', '.join(items)
+                else:
+                  val = items
+              if val is not None and \
+                 val.startswith('{$uid:') and val.endswith('}'):
+                obj = self.getLinkObj(val, REQUEST)
+                if obj is not None:
+                  try:
+                    lang = REQUEST.get('lang', self.getPrimaryLanguage())
+                    if ';lang=' in val:
+                      lang = self.re_sub(r'{\$(.*);lang=(\w*)}', r'\2', val)
+                    val = self.UniBE_zeix_('getHref2SubdomainHtml', item=obj, lang=lang)
+                  except AttributeError:
+                    val = obj.getHref2IndexHtml(REQUEST)
+              values[key] = val
             rows.append(values)
 
         csvfile = io.StringIO()
