diff --git a/Products/zms/zpt/ZMS/manage_customize.zpt b/Products/zms/zpt/ZMS/manage_customize.zpt
index 6bdeacc3..41761fc9 100644
--- a/Products/zms/zpt/ZMS/manage_customize.zpt
+++ b/Products/zms/zpt/ZMS/manage_customize.zpt
@@ -37,7 +37,7 @@
 							<option value="">Select type to add...</option>
 							<tal:block tal:repeat="meta_type meta_types"
 								><option tal:condition="python:len(here.objectValues(meta_type['name']))==0"
-									tal:attributes="value meta_type/action" 
+									tal:attributes="value meta_type/action"
 									tal:content="python:here.getZMILangStr(meta_type['name'])">name</option>
 							</tal:block>
 						</select>
@@ -255,7 +255,7 @@
 							<span class="zmi-code pt-1" tal:condition="python:conf_key.endswith('.password')">******</span>
 						</td>
 					</tr>
-				</tbody>	
+				</tbody>
 			</table>
 			</tal:block>
 		</form>
@@ -335,10 +335,7 @@
 				<div class="form-group row">
 					<label class="col-sm-2 control-label">ZMS Version</label>
 					<div class="col-sm-10">
-						<a title="ZMS commits on github.com" target="_blank"
-							href="https://github.com/zms-publishing/ZMS/commits/main">
-							<samp id="zms_custom_version" tal:content="structure python:here.zms_version(custom=True)">zms_version</samp>
-						</a>
+                        <samp tal:content="structure python:here.zms_version(custom=True)">zms_version</samp>
 					</div>
 				</div><!-- .form-group -->
 				<div class="form-group row">
@@ -363,22 +360,21 @@
 					<label class="col-sm-2 control-label" for="installed_python_packages">
 						Installed Python Packages
 						<p class="help-text small mt-sm-3 mb-0">
-							Tools such as the <a href="https://pip.pypa.io/en/stable/reference/pip/" target="_blank">Python Package Installer</a>
+							Tools such as the <a href="https://pip.pypa.io/en/stable/cli/pip/" target="_blank">Python Package Installer</a>
 							with <a href="https://pip.pypa.io/en/stable/user_guide/#requirements-files" target="_blank">requirements</a>
 							and <a href="https://pip.pypa.io/en/stable/user_guide/#constraints-files" target="_blank">constraints</a> files
-							and <a href="https://requires.io/github/zopefoundation/Zope/requirements/?branch=master" target="_blank">requires.io</a>
 							help manage and monitor package dependencies.
 						</p>
 					</label>
 					<div class="col-sm-10">
 						<div class="custom-control custom-radio custom-control-inline">
-							<input type="radio" class="custom-control-input" 
+							<input type="radio" class="custom-control-input"
 								onchange="show_pip_cmd($(this))"
 								id="pip_cmd1" name="pip_cmd" value="freeze" checked="checked" />
 							<label class="custom-control-label" for="pip_cmd1"><samp>pip freeze --all</samp></label>
 						  </div>
 						  <div class="custom-control custom-radio custom-control-inline">
-							<input type="radio" class="custom-control-input" 
+							<input type="radio" class="custom-control-input"
 								onchange="show_pip_cmd($(this))"
 								id="pip_cmd2" name="pip_cmd" value="list" />
 							<label class="custom-control-label" for="pip_cmd2"><samp>pip list</samp></label>
