diff --git a/Products/zms/_confmanager.py b/Products/zms/_confmanager.py
--- a/Products/zms/_confmanager.py	(revision 7e8ca7aab4027a3dd36dcb9bb5552758fd331fdd)
+++ b/Products/zms/_confmanager.py	(date 1712821693479)
@@ -480,7 +480,9 @@
         {'key':'ZMS.pathhandler.id_quote.mapping','title':'Declarative IDs-Mapping','desc':'ZMS can map characters in DC.Title.Alt to declarative IDs.','datatype':'string','default':' _-_/_'},
         {'key':'ZMS.preview.contentEditable','title':'Content-Editable Preview','desc':'Make content in ZMS preview editable','datatype':'boolean','default':1},
         {'key':'ZMS.pathcropping','title':'Crop URLs','desc':'ZMS can crop the SERVER_NAME from URLs.','datatype':'boolean'},
-        {'key':'ZMS.manage_tabs_message','title':'Global Message','desc':'ZMS can display a global message for all users in the management interface.','datatype':'text'},
+        {'key':'ZMS.manage_tabs_message','title':'Global Message: Success (green)','desc':'ZMS can display a global message for all users in the management interface.','datatype':'text'},
+        {'key':'ZMS.manage_tabs_warning_message','title':'Global Message: Warning (yellow)','desc':'ZMS can display a global message for all users in the management interface.','datatype':'text'},
+        {'key':'ZMS.manage_tabs_danger_message','title':'Global Message: Danger (red)','desc':'ZMS can display a global message for all users in the management interface.','datatype':'text'},
         {'key':'ZMS.http_accept_language','title':'Http Accept Language','desc':'ZMS can use the HTTP_ACCEPT_LANGUAGE request-parameter to determine initial language.','datatype':'boolean'},
         {'key':'ZMS.export.domains','title':'Export resources from external domains','desc':'ZMS can export resources from external domains in the HTML export.','datatype':'string'},
         {'key':'ZMS.export.pathhandler','title':'Export XHTML with decl. Document Ids','desc':'Please activate this option, if you would like to generate declarative document URLs for static XHTML-Export: /documentname/index_eng.html will be transformed to /documentname.html','datatype':'boolean'},
@@ -564,6 +566,12 @@
       params = ('key', 'default', 'REQUEST')
       [operator.setitem(kwargs, params[x], args[x]) for x in range(len(args))]
       key = kwargs['key']
+      # override value of system property
+      # with value of corresponding environment variable if set
+      if key is not None:
+        env_key = key.upper().replace('.', '_')
+        if env_key in os.environ:
+          return(os.getenv(env_key))
       default = kwargs.get('default')
       REQUEST = kwargs.get('REQUEST')
       if REQUEST is not None:
