diff --git a/Products/zms/zmsindex.py b/Products/zms/zmsindex.py
index 1f766f18..3e4fc9ab 100644
--- a/Products/zms/zmsindex.py
+++ b/Products/zms/zmsindex.py
@@ -270,7 +270,7 @@ class ZMSIndex(ZMSItem.ZMSItem):
       
       log = []
       catalog = self.get_catalog()
-      regenerate_all = standard.pybool(regenerate_all) or len(catalog) == 0
+      regenerate_all = standard.pybool(regenerate_all)
       regenerate_duplicates = standard.pybool(regenerate_duplicates) or regenerate_all
       urls = [x for x in REQUEST.get('url').split(',') if x] if REQUEST else ['{$}']
       for url in urls:
