diff --git a/Products/zms/_objinputs.py b/Products/zms/_objinputs.py
index 9937c335..eb88b5fc 100644
--- a/Products/zms/_objinputs.py
+++ b/Products/zms/_objinputs.py
@@ -159,7 +159,7 @@ class ObjInputs(object):
   #	@param btn    Appear as Bootstrap Button
   #	@return String
   #
-  # Hint: In order to avoid multiple identical ids (HTML5 conformance) 
+  # Hint: In order to avoid multiple identical ids (HTML5 conformance)
   # call getCheckbox(..., elId='', ...) with an empty string for elId
   # ----------------------------------------------------------------------------
   def getCheckbox(self, fmName, elName, elId=None, value=None, enabled=True, hidden=True, css='', btn=False, options=[0, 1]):
@@ -186,7 +186,7 @@ class ObjInputs(object):
     html.append(' type="checkbox"')
     if not enabled:
       html.append(' disabled="disabled"')
-    if checked: 
+    if checked:
       html.append(' checked="checked"')
     html.append(' onclick="if(this.checked){$(this)%s.prev().val(\'%s\')}else{$(this)%s.prev().val(\'%s\')}"'%(['', '.parent()'][btn], options[1], ['', '.parent()'][btn], options[0]))
     html.append(' />')
