diff --git a/Products/zms/zpt/ZMSRecordSet/grid.zpt b/Products/zms/zpt/ZMSRecordSet/grid.zpt
index 21a2e9a3..c8ce293a 100644
--- a/Products/zms/zpt/ZMSRecordSet/grid.zpt
+++ b/Products/zms/zpt/ZMSRecordSet/grid.zpt
@@ -7,6 +7,7 @@
 		metaObjIds python:here.getMetaobjIds();
 		metaObjAttrIds python:here.getMetaobjAttrIds(metaObj['id']);
 		metaObjAttrs python:metaObj['attrs'][1:];
+		metaObjAttrs python:[x for x in metaObjAttrs if x.get('custom', 0) == 1];
 		num_rows python:here.attr('_num_rows');
 		command python:'Save';
 		dummy0 python:standard.once('zmiGraphicExtEdit',request);
@@ -152,7 +153,8 @@
 			<div class="card-body">
 				<div class="form-group row">
 					<div class="controls save">
-						<button type="submit" name="btn" class="btn btn-primary" tal:attributes="value python:'BTN_%s'%command.upper()" tal:content="python:here.getZMILangStr('BTN_%s'%command.upper())">Action</button>
+						<button type="submit" name="btn" class="btn btn-primary" tal:attributes="value python:'BTN_%s'%command.upper()" tal:content="python:here.getZMILangStr('BTN_%s'%command.upper())"
+                            tal:condition="python:here.getPrimaryLanguage()==request.get('lang')">Action</button>
 						<button type="submit" name="btn" class="btn btn-secondary" value="BTN_CANCEL" tal:content="python:here.getZMILangStr('BTN_CANCEL')">Cancel</button>
 					</div>
 				</div><!-- .form-row -->
