diff --git a/Products/zms/zmsobject.py b/Products/zms/zmsobject.py
index 3dc4cef8..c619069f 100644
--- a/Products/zms/zmsobject.py
+++ b/Products/zms/zmsobject.py
@@ -624,7 +624,7 @@ class ZMSObject(ZMSItem.ZMSItem,
               title += '; '+'; '.join(['RESTRICTION: '+x[1] for x in constraints['RESTRICTIONS']])
       else:
         name = 'fas fa-exclamation-triangle constraint-error'
-        title = '%s not found!'%str(bool(meta_id) or 'Meta-ID')
+        title = '%s not found!'%str(bool(meta_id!='' and meta_id  or 'Meta-ID') or 'Meta-ID')
       return '<i class="%s" title="%s"%s></i>'%(name,title,extra)
 
 
