diff --git a/Products/zms/ZMSItem.py b/Products/zms/ZMSItem.py
index c4f4d483..aa308e25 100644
--- a/Products/zms/ZMSItem.py
+++ b/Products/zms/ZMSItem.py
@@ -43,9 +43,9 @@ class ZMSItem(
 
     # Documentation string.
     __doc__ = """ZMS product module."""
-    # Version string. 
-    __version__ = '0.1' 
-    
+    # Version string.
+    __version__ = '0.1'
+
     # Management Permissions.
     # -----------------------
     __viewPermissions__ = (
@@ -113,7 +113,7 @@ class ZMSItem(
         if request['ZMS_ROOT'].startswith(base):
           request.set( 'ZMS_ROOT', request['ZMS_ROOT'][len(base):])
           request.set( 'ZMS_COMMON', request['ZMS_COMMON'][len(base):])
-    
+
     def zmi_page_request(self, *args, **kwargs):
       request = self.REQUEST
       RESPONSE = request.RESPONSE
@@ -130,6 +130,10 @@ class ZMSItem(
         request.set('lang', langs[0])
       if not request.get('manage_tabs_message'):
         request.set( 'manage_tabs_message', self.getConfProperty('ZMS.manage_tabs_message', ''))
+      if not request.get('manage_tabs_warning_message'):
+        request.set( 'manage_tabs_warning_message',self.getConfProperty('ZMS.manage_tabs_warning_message',''))
+      if not request.get('manage_tabs_danger_message'):
+        request.set( 'manage_tabs_danger_message',self.getConfProperty('ZMS.manage_tabs_danger_message',''))
       if 'zmi-manage-system' in request.form:
         standard.set_session_value(self,'zmi-manage-system',int(request.get('zmi-manage-system',0)))
       # AccessableObject
